projects/plugins/crm/src/automation/commons/triggers/contacts/class-contact-deleted.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Jetpack CRM Automation Contact_Deleted trigger.
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Triggers;
+
+use Automattic\Jetpack\CRM\Automation\Automation_Exception;
+use Automattic\Jetpack\CRM\Automation\Automation_Workflow;
+use Automattic\Jetpack\CRM\Automation\Base_Trigger;
+
+/**
+ * Adds the Contact_Deleted class.
+ */
+class Contact_Deleted extends Base_Trigger {
+
+	/**
+	 * Contructs the Contact_Deleted instance.
+	 */
+	public function __construct() {
+		$trigger_data = array(
+			'name'        => 'contact_delete',
+			'title'       => 'Contact Deleted',
+			'description' => 'Triggered when a contact is deleted.',
+			'category'    => 'contact',
+		);
+
+		parent::__construct( $trigger_data );
+	}
```

Comment:
In the main branch,[ I replied to this one](https://github.com/Automattic/jetpack/pull/30366#discussion_r1201254426). Really we don't need to call the parent constructor, in fact, we can instance triggers without any property filled. The Automation System will be responsible for accepting triggers or not, for example, we only accept triggers with names.

That said, each trigger implementation should set its own properties directly:

```
public function __construct() {
    $this->name         = 'contact_delete';
    $this->title        = __( 'Contact Deleted', 'zero-bs-crm' );
    $this->description  = __( 'Triggered when a contact is deleted.', 'zero-bs-crm' );
    $this->category     = 'contact';
}
```

or... since the trigger data are immutable (set them programmatically), using the static way that will facilitate access to the trigger data information without the need to instance it:

```
public function __construct() {
    $self::name         = 'contact_delete';
    $self::title        = __( 'Contact Deleted', 'zero-bs-crm' );
    $self::description  = __( 'Triggered when a contact is deleted.', 'zero-bs-crm' );
    $self::category     = 'contact';
} 
```